python/ray/tests/test_gpu_objects_gloo.py

```diff
@@ -422,5 +426,99 @@ def test_tensor_extracted_from_tensordict_in_gpu_object_store(ray_start_regular)
     assert torch.equal(ret_val_src[1], td["reward"])
 
 
+def test_duplicate_objectref_transfer(ray_start_regular):
+    """
+    This test checks for a race condition when the same GPU object ref 
+    is passed to the same actor multiple times.
+    
+    Expected behavior: Each call should get its own copy of the tensor.
+    Bug: Background receives may overwrite each other before pops happen.
+    """
+    world_size = 2
+    actors = [GPUTestActor.remote() for _ in range(world_size)]
+    create_collective_group(actors, backend="torch_gloo")
+
+    actor0, actor1 = actors[0], actors[1]
+
+    small_tensor = torch.randn((1,))
+    
+    # Store the original value for comparison
+    original_value = small_tensor.clone()
```

Comment:
`clone()` isn't necessary since it will copy anyway when it's sent to the actor0 process.